pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target ottl.PSliceGetter[K]
+	Item   string
```

Comment:
Can we have `Contains.item` as string type? If not, I will try to fix it.

Reply:
Could we just use an untyped `ottl.Getter` here?

Reply:
This is what I missed. Thanks!